### PR TITLE
Test: Zero-amount invoices and debit-notes are settled automatically on acceptance

### PR DESCRIPTION
**Pull request by pnowosie**
Uses `requestor.wait_for_approval(agreement_id)` and `requestor.terminate_agreement(agreement_id, None)` to make this test as quick as possible.

Not implemented (due `goth` lack of support):
- no debit notes check
- no provider invoice check

### CODE REVIEW

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (pnowosie) [inline test/yagna/module/payments/test_zero_amount_txs.py:121]:
```suggestion
        #  Zero-amount invoice is issued when agreement is terminated without activity
```

**Comment by maaktweluit** [comment]
Changed the base to the PR you rebased onto, so it shows the diff better
Once that branch is merged it automagically will change this PR's target too

LGTM!

Guess the failing build is related to the sub-pr, maybe its not compatible with `master`..
Will approve when its all 💚

**Review by Wiezzel — APPROVED** [review]
Looks fine.

**Comment by pnowosie** [comment]
@zakaprov - am I free to merge just with codestyle check?
(it is again based back on `master` to not wait for #348)

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (zakaprov) [inline test/yagna/module/payments/test_zero_amount_txs.py:128]: The `sleep` here is not acceptable. We don't want this much idle overhead on our CI test runs.

Instead, I'd suggest to just add the proper `terminate_agreement` implementation. You can add these two methods to their respective classes, they will be analogous to what's already there. For `requestor.terminate_agreement` take a look at:
https://github.com/golemfactory/goth/blob/adf441cbe07e1bbf8a9c44b10ba86ef103455064/goth/runner/requestor.py#L166-L169
Just create a new function and replace `confirm_agreement` with `terminate_agreement` in the client call.
For the provider log assertion:
https://github.com/golemfactory/goth/blob/adf441cbe07e1bbf8a9c44b10ba86ef103455064/goth/runner/provider.py#L37-L40
Add a new function like the one above and replace the log regex with the one you need.

**Review by zakaprov — CHANGES_REQUESTED** [review]
Please replace the `sleep` with a proper agreement termination, the implementation will be analogous to already existing functions.

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (Wiezzel) [inline test/yagna/module/payments/test_zero_amount_txs.py:154]: How about that?
```suggestion
    deadline = time.time() + 30
    while True:
        assert time.time() < deadline, "Retries time exceeded"
```

Comment (pnowosie) [reply test/yagna/module/payments/test_zero_amount_txs.py:154]: I think I will name it `timeout_seconds`, but I like an idea!

**Review by Wiezzel — APPROVED** [review]

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (Wiezzel) [reply test/yagna/module/payments/test_zero_amount_txs.py]: I think "dealine" would be more appropriate because it's a _point in time_ while "timeout" is used to refer to a _period of time_ but let's not be fussy.

Comment (pnowosie) [reply test/yagna/module/payments/test_zero_amount_txs.py]: UPDATE: Mentioned code is gone as I try `requestor.wait_for_approval(agreement_id)`

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (zakaprov) [inline test/yagna/module/payments/test_zero_amount_txs.py:99]: I don't think we need two log messages one after another here, I'd remove this one.

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (zakaprov) [inline test/yagna/module/payments/test_zero_amount_txs.py:103]: This _is_ required, otherwise the test fails. No need to leave a comment here I think.

File: test/yagna/module/payments/test_zero_amount_txs.py
Comment (zakaprov) [inline test/yagna/module/payments/test_zero_amount_txs.py:142]: Let's not leave this block hanging here, I've added an issue (https://github.com/golemfactory/goth/issues/433) to track progress.

**Review by zakaprov — APPROVED** [review]
Looks good, please address my comments before merging.

**Pull request merged**
